src/authentication/authentication-hapi-plugin.ts

```diff
@@ -106,9 +94,37 @@ class AuthenticationHapiPlugin extends HapiPlugin {
 
   public async getTeamTokenHandler(request: Hapi.Request, reply: Hapi.IReply) {
     try {
-      const teamId = parseInt(request.paramsArray[0], 10);
-      const teamToken: TeamToken[] = await teamTokenQuery(this.db, undefined, teamId);
-      if (!teamToken.length) {
+      const credentials = request.auth.credentials as AccessToken;
+      const userName = sanitizeUsername(credentials.sub);
+      const isAdmin = await this.isAdmin(userName);
+      const userTeams = await this.gitlab.getUserGroups(userName);
+      const parameterKey = 'teamIdOrName';
```

Comment:
What I meant was that since it's defined and used straight away, maybe this is enough:
```javascript
const teamIdOrName = request.params['teamIdOrName'];
```